sphinx/builders/linkcheck.py

```diff
@@ -98,74 +115,75 @@ def process_result(self, result: CheckResult) -> None:
-            if result.message:
-                logger.info(darkgray('-ignored- ') + result.uri + ': ' + result.message)
-            else:
-                logger.info(darkgray('-ignored- ') + result.uri)
-        elif result.status == 'local':
-            logger.info(darkgray('-local-   ') + result.uri)
-            self.write_entry('local', result.docname, filename, result.lineno, result.uri)
-        elif result.status == 'working':
-            logger.info(darkgreen('ok        ') + result.uri + result.message)
-        elif result.status == 'timeout':
-            if self.app.quiet or self.app.warningiserror:
-                logger.warning('timeout   ' + result.uri + result.message,
-                               location=(result.docname, result.lineno))
-            else:
-                logger.info(red('timeout   ') + result.uri + red(' - ' + result.message))
-            self.write_entry('timeout', result.docname, filename, result.lineno,
-                             result.uri + ': ' + result.message)
-            self.timed_out_hyperlinks += 1
-        elif result.status == 'broken':
-            if self.app.quiet or self.app.warningiserror:
-                logger.warning(__('broken link: %s (%s)'), result.uri, result.message,
-                               location=(result.docname, result.lineno))
-            else:
-                logger.info(red('broken    ') + result.uri + red(' - ' + result.message))
-            self.write_entry('broken', result.docname, filename, result.lineno,
-                             result.uri + ': ' + result.message)
-            self.broken_hyperlinks += 1
-        elif result.status == 'redirected':
-            try:
-                text, color = {
-                    301: ('permanently', purple),
-                    302: ('with Found', purple),
-                    303: ('with See Other', purple),
-                    307: ('temporarily', turquoise),
-                    308: ('permanently', purple),
-                }[result.code]
-            except KeyError:
-                text, color = ('with unknown code', purple)
-            linkstat['text'] = text
-            if self.config.linkcheck_allowed_redirects:
-                logger.warning('redirect  ' + result.uri + ' - ' + text + ' to ' +
-                               result.message, location=(result.docname, result.lineno))
-            else:
-                logger.info(color('redirect  ') + result.uri +
-                            color(' - ' + text + ' to ' + result.message))
-            self.write_entry('redirected ' + text, result.docname, filename,
-                             result.lineno, result.uri + ' to ' + result.message)
-        else:
-            raise ValueError('Unknown status %s.' % result.status)
+
+        match result.status:
+            case LinkStatus.IGNORED:
+                msg = darkgray('-ignored- ') + result.uri + f': {result.message}' if result.message else ''  # NoQA: E501
```

Comment:
As hinted by the appearance of `E501`: some of these output message constructions are a bit lengthy.  I'm considering extracting a small helper function to attempt to wrap the intended behaviour.